pkg/controller/persistence/dgraph/service.go

```diff
@@ -0,0 +1,37 @@
+package dgraph
+
+import (
+	"encoding/json"
+	"time"
+
+	api_v1 "k8s.io/api/core/v1"
+)
+
+type Service struct {
+	ID
+	IsService string    `json:"isService"`
+	Name      string    `json:"name,omitempty"`
+	StartTime time.Time `json:"startTime,omitempty"`
+	EndTime   time.Time `json:"endTime,omitempty"`
+	Pod       []*Pod    `json:"servicePods,omitempty"`
+}
+
+func PersistService(service api_v1.Service) error {
+	xid := service.Namespace + ":" + service.Name
+	uid, _ := GetUId(Client, xid, "isService")
+
+	if uid == "" {
+		// If pod is not present, persist it.
+		newService := Service{
```

Comment:
same as comment in `dgraph/pod.go` what about other fields `StartTime`?

Reply:
WIP, I will incorporate in future commits.